src/internal/common/errors.ts

```diff
@@ -10,14 +18,30 @@ export interface ErrorProps {
   [key: string]: unknown;
 }
 
+
 /**
  * Creates a ShipEngine Connect SDK error
  */
-export function error(code: ErrorCode, message: string, { originalError, ...props }: ErrorProps = {}): AppError {
-  // Create a new error with:
-  //  - The new error message and the original error message
-  //  - The new stack trace and the original stack trace
-  //  - The new properties and the original error's properties
-  const err = ono(originalError as Error, { ...props, code }, message);
-  return err;
+export function error(code: ErrorCode | SystemErrorCode, message: string, props: ErrorProps = {}): AppError {
```

Comment:
I like that the SDK wraps all errors and automatically assigns codes to the internal error types, but it seems weird to me that this internal function returns an `AppError` since that means it has to conform to:
```
code: ErrorCode | string;
```
I think it would be more useful if there was an internal error type that had a more strict code:
```
code: ErrorCode | SystemErrorCode;
```
So that platform consumers would know all of the enum values to expect.

Reply:
Good suggestion.  I created a `SystemError` interface 👍